File: 566-reshape-the-matrix/566-reshape-the-matrix.js
/**
 * @param {number[][]} mat
 * @param {number} r
 * @param {number} c
 * @return {number[][]}
 */
var matrixReshape = function(mat, r, c) {
    let m = mat.length;
    let n = mat[0].length;
    let inner = [];
    let outer = [];
    if (m*n !== r*c){
        return mat;
    } else {
        for(let i=0; i<m; i++){
            for (let j=0; j<n; j++){
                if (inner.length < c){
                    inner.push(mat[i][j]);
                } else {
                    outer.push(inner);
                    inner = [];
                    inner.push(mat[i][j]);
                }
            }
        }
        if (outer.length < r){
            outer.push(inner);
        }
        return outer;
    }
};

File: 566-reshape-the-matrix/566-reshape-the-matrix.py
class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        m = len(mat)
        n = len(mat[0])
        inner = []
        outer = []
        print(m*n)
        print(r*c)
        if (m*n) != (r*c):
            return mat
        else:
            for i in range(m):
                for j in range(n):
                    if len(inner) < c:
                        inner.append(mat[i][j])
                    else:
                        outer.append(inner)
                        inner = []
                        inner.append(mat[i][j])
            if len(outer) != r:
                outer.append(inner)
            return outer